drivers/lorabee/lorabee.c

```diff
@@ -0,0 +1,214 @@
+        return -LORABEE_ERR_INVALID_PARAM;
+    }
+    else if (strcmp((char*)dev->resp_buf, "not_joined") == 0) {
+        DEBUG("[lorabee] failed: network is not joined\n");
+        return -LORABEE_ERR_NOT_JOINED;
+    }
+    else if (strcmp((char*)dev->resp_buf, "no_free_ch") == 0) {
+        DEBUG("[lorabee] failed: all channels are busy\n");
+        return -LORABEE_ERR_NO_FREE_CH;
+    }
+    else if (strcmp((char*)dev->resp_buf, "silent") == 0) {
+        DEBUG("[lorabee] failed: all channels are busy\n");
+        return -LORABEE_ERR_SILENT;
+    }
+    else if (strcmp((char*)dev->resp_buf,
+             "frame_counter_err_rejoin_needed") == 0) {
+        DEBUG("[lorabee] failed: Frame counter rolled over\n");
+        return -LORABEE_ERR_FR_CNT_REJOIN_NEEDED;
+    }
+    else if (strcmp((char*)dev->resp_buf, "busy") == 0) {
+        DEBUG("[lorabee] failed: MAC state is in an Idle state\n");
+        return -LORABEE_ERR_BUSY;
+    }
+    else if (strcmp((char*)dev->resp_buf, "invalid_data_len") == 0) {
+        DEBUG("[lorabee] failed: payload length too large\n");
+        return -LORABEE_ERR_INVALID_DATA_LENGTH;
+    }
+
+    return LORABEE_OK;
+}
+
+/*
+ * Interrupt callbacks
+ */
+static void _rx_cb(void *arg, uint8_t c)
+{
+    lorabee_t *dev = (lorabee_t *)arg;
+
+    /* The device sends \r\n when done */
+    if (c == '\r') {
+        if (dev->resp_size != 0) {
+            dev->resp_buf[dev->resp_size++] = '\0';
+        }
+        dev->resp_done = 1;
+    }
+    else if (c == '\n') {
+        return;
+    }
+    else {
+        dev->resp_buf[dev->resp_size++] = c;
+    }
+
+    mutex_unlock(&(dev->resp_lock));
+}
+
+/*
+ * Driver's "public" functions
+ */
+void lorabee_setup(lorabee_t *dev, const lorabee_params_t *params)
+{
+    assert(dev && (params->uart < UART_NUMOF));
+
+    /* set peripherals to use */
+    memcpy(&dev->p, params, sizeof(lorabee_params_t));
+
+    /* initialize pins */
+    if (dev->p.pin_reset != GPIO_UNDEF) {
+        gpio_init(dev->p.pin_reset, GPIO_OUT);
+        gpio_set(dev->p.pin_reset);
+    }
+    /* we initialize the UART later, since we can not handle interrupts, yet */
+}
+
+int lorabee_init(lorabee_t *dev)
+{
+    /* initialize buffers and locks*/
+    mutex_init(&(dev->cmd_lock));
+    dev->resp_size = 0;
+
+    /* initialize UART and GPIO pins */
+    if (uart_init(dev->p.uart, dev->p.baudrate, _rx_cb, dev) != UART_OK) {
+        DEBUG("[lorabee] init: error initializing UART\n");
+        return -ENXIO;
+    }
+
+    /* if reset pin is connected, do a hardware reset */
+    if (dev->p.pin_reset != GPIO_UNDEF) {
+        gpio_clear(dev->p.pin_reset);
+        xtimer_usleep(RESET_DELAY);
+        gpio_set(dev->p.pin_reset);
+    }
+
+    if (lorabee_write_cmd(dev, (uint8_t*)"") < 0) {
+        return -1;
+    }
+
+    DEBUG("[lorabee] init: Initialization successful\n");
+    return 0;
+}
+
+int lorabee_write_cmd(lorabee_t *dev, uint8_t *cmd)
```

Comment:
I'd prefer that cmd is a const char*. Every command you send to the RN device is ASCII. No need to pretend it can be binary too. Only where uart_write is being called you need a type cast.